spec/enumerables_spec.rb

```diff
@@ -0,0 +1,273 @@
+        expect(array3.my_any?(nil)).to be true
+        expect(array2.my_any?(/o/)).to be true
+      end
+
+      it 'should return false if pattern doesn\'t match any element in collection' do
+        expect(array2.my_any?(Integer)).to be false
+        expect(array2.my_any?(/s/)).to be false
+      end
+    end
+  end
+
+  describe '#my_none?' do
+    context 'when no block is given' do
+      context 'if none of the elements is not (false) or (nil)' do
+        it 'should return true' do
+          expect(array4.my_none?).to be true
+        end
+      end
+
+      context 'if  atleast one of elements is not (false) or (nil)' do
+        it 'should return false' do
+          expect(array3.my_none?).to be false
+        end
+      end
+    end
+
+    context 'when block given' do
+      it 'should return true if none of the collection items evaluates to true' do
+        expect(array5.my_none? { |ele| ele == 4 }).to be true
+        expect(range.my_none? { |ele| ele > 5 }).to be true
+      end
+
+      it 'should return false if any of the collection items evaluates to true' do
+        expect(array5.my_none? { |ele| ele == 3 }).to be false
+        expect(range.my_none? { |ele| ele > 4 }).to be false
+      end
+    end
+
+    context 'when pattern is given as argument' do
+      it 'should return true if pattern doesn\'t match any element in collection' do
+        expect(array3.my_none?(String)).to be true
+        expect(array2.my_none?(/s/)).to be true
+      end
+
+      it 'should return false if pattern === any element in collection' do
+        expect(array3.my_none?(nil)).to be false
+        expect(array2.my_none?(/o/)).to be false
+      end
+    end
+  end
+
+  describe '#my_count' do
+    context 'when block given' do
+      it 'counts number of collection items yielding a true value' do
+        expect(array5.my_count { |ele| ele == 2 }).to eq(3)
+        expect(array1.my_count { |ele| ele > 2 }).to eq(2)
+      end
+    end
+
+    context 'when no block is given' do
+      it 'should return number of items in collection' do
+        expect(array2.count).to eq(4)
+        expect(array4.count).to eq(4)
+      end
+    end
+
+    context 'when argument is given' do
+      it 'should return number of items in collection that are equal to argument' do
+        expect(array5.my_count(2)).to eq(3)
+      end
+    end
+  end
+
+  describe '#my_map' do
+    context 'when block given' do
+      it 'should return a new array with results of running block once for every collection item' do
+        expect(array1.my_map { |ele| ele + 1 }).to eq([2, 3, 4, 5])
+        expect(range.my_map { |ele| ele + 2 }).to eq([3, 4, 5, 6, 7])
+        expect(range.my_map { 'cat' }).to eq(%w[cat cat cat cat cat])
+      end
+    end
+
+    context 'when no block is given' do
+      it 'should return an Enumerator ' do
+        expect(array1.my_map).to be_a_kind_of(Enumerator)
+        expect(array1.my_map.to_a).to eq(array1)
```

Comment:
* [ ] You should have one `it` block for each expected behaviour. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/itblock.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))